src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -255,54 +255,58 @@ public static function get_user_data_from_wp_global_styles( $theme, $create_post
 		}
 		$user_cpt         = array();
 		$post_type_filter = 'wp_global_styles';
+		$stylesheet       = $theme->get_stylesheet();
 		$args             = array(
-			'numberposts' => 1,
-			'orderby'     => 'date',
-			'order'       => 'desc',
-			'post_type'   => $post_type_filter,
-			'post_status' => $post_status_filter,
-			'tax_query'   => array(
+			'posts_per_page'      => 1,
+			'orderby'             => 'post_date',
+			'order'               => 'desc',
+			'post_type'           => $post_type_filter,
+			'post_status'         => $post_status_filter,
+			'ignore_sticky_posts' => true,
+			'no_found_rows'       => true,
+			'tax_query'           => array(
 				array(
 					'taxonomy' => 'wp_theme',
 					'field'    => 'name',
-					'terms'    => $theme->get_stylesheet(),
+					'terms'    => $stylesheet,
 				),
 			),
 		);
 
 		$cache_key = sprintf( 'wp_global_styles_%s', md5( serialize( $args ) ) );
-		$post_id   = wp_cache_get( $cache_key );
-
-		if ( (int) $post_id > 0 ) {
-			return get_post( $post_id, ARRAY_A );
-		}
-
+		$post_id   = (int) get_transient( $cache_key );
 		// Special case: '-1' is a results not found.
 		if ( -1 === $post_id && ! $create_post ) {
 			return $user_cpt;
 		}
 
-		$recent_posts = wp_get_recent_posts( $args );
-		if ( is_array( $recent_posts ) && ( count( $recent_posts ) === 1 ) ) {
-			$user_cpt = $recent_posts[0];
+		if ( $post_id > 0 && in_array( get_post_status( $post_id ), (array) $post_status_filter, true ) ) {
```

Comment:
> If the post is deleted in the database, then this will return null.

This is unintentionally covered by the `in_array( get_post_status( $post_id ), (array) $post_status_filter, true ) )` check. If the post is deleted then `false` will not be in the array.

I think it would be good to intentionally handle this.

```suggestion
		if (
			$post_id > 0
			&& get_post( $post_id )
			&& in_array( get_post_status( $post_id ), (array) $post_status_filter, true )
		) {
```

Reply:
So I thought of this. We don't need to a get_post here, as that is the first line of `get_post_status`. If the post is delete, that function will return false.

Reply:
Yeah, I get that but I think `get_post()` will be good as a defensive coding technique.

* `get_post()` serves an explicit purpose: to check that the post exists
* `get_post_status()` checks the status but works to check the post exists by coincidence.

If a future dev decides the `get_post_status()` check isn;t needed due to the `post_status` setting in query, then a bug is introduced.

As the very least it requires a detailed comment explaining that `get_post_status` serves two purposes.